airflow/models/dagrun.py

```diff
@@ -201,7 +201,7 @@ def get_task_instances(self, state=None, session=None):
 
         if self.dag and self.dag.partial:
             tis = tis.filter(TaskInstance.task_id.in_(self.dag.task_ids))
-        return tis.all()
+        return tis.order_by(TaskInstance.execution_date, TaskInstance.dag_id, TaskInstance.task_id).all()
```

Comment:
How about changing the test itself so order doesn't matter?

Reply:
Good point. That's done. I changed the test instead.